lib/oli/delivery/metrics.ex

```diff
@@ -32,20 +44,28 @@ defmodule Oli.Delivery.Metrics do
           dynamic([cp, _], cp.container_id == ^container_id)
       end
 
+    pages_count =
+      from(cp in ContainedPage)
+      |> where([cp], cp.section_id == ^section_id)
+      |> where(^filter_by_container)
+      |> select([cp], count(cp.id))
+      |> Repo.one()
+      |> max(1)
+
     query =
       ContainedPage
-      |> join(:left, [cp], ra in ResourceAccess, on: cp.page_id == ra.resource_id and cp.section_id == ra.section_id and ra.user_id == ^user_id)
+      |> join(:left, [cp], ra in ResourceAccess,
```

Comment:
Since you're using a `left join` here, if you have a user that doesn't have a resource access for a given contained page, it will still take that contained page into account which will add a nil entry in the returned map (this just happened to me `%{2 => 0.75, nil => nil}`. Changing it to an `inner join` should do the trick 😁

Reply:
Makes sense, good catch, thanks!